Update OTel Spring starter doc with 2.6.0 features (#4847)
open-telemetry/opentelemetry.io: content/en/docs/zero-code/java/spring-boot-starter/out-of-the-box-instrumentation.md

| Micrometer            | `otel.instrumentation.micrometer.enabled`       | false         |
| R2DBC (reactive JDBC) | `otel.instrumentation.r2dbc.enabled`            | true          |

> ## Suppressing specific instrumentation

Reviewer: ```suggestion
## Turn on instrumentations selectively
```

> To enable only specific instrumentations, you can disable all the
> instrumentations by setting the `otel.instrumentation.common.default-enabled`
> property to `false`, and enable instrumentations one by one. For example, if you
> want to only enable the JDBC instrumentation, you can set
> `otel.instrumentation.jdbc.enabled` to `true`.

Reviewer: ```suggestion
To use only specific instrumentations, turn off all the instrumentations
first by setting the `otel.instrumentation.common.default-enabled`
property to `false`. Then, turn on instrumentations one by one. 

For example, if you want to only enable the JDBC instrumentation,
set `otel.instrumentation.jdbc.enabled` to `true`.
```

## Common instrumentation configuration

> To enable only specific instrumentations, you can disable all the

Reviewer 2: - should be part of previous section 
- section should have a title, e.g. "Suppressing specific instrumentation"
- section needs an explanation like here: https://opentelemetry.io/docs/zero-code/java/agent/configuration/#suppressing-specific-agent-instrumentation

Author: Change done as dicussed during our chat

}
```

As it's possible with the
[Java agent](/docs/zero-code/java/agent/configuration), you can configure the
capture of:

> - HTTP request and response headers

Reviewer 2: this will move to a different page with https://github.com/open-telemetry/opentelemetry.io/pull/4811

}
```

As it's possible with the

> [Java agent](/docs/zero-code/java/agent/configuration), you can configure the

Reviewer 2: ```suggestion
Java agent, you can configure the
```

> capture of:

Reviewer: ```suggestion
capture of the following entities:
```

> - HTTP request and response headers

Reviewer 2: ```suggestion
- [HTTP request and response headers](https://opentelemetry.io/docs/zero-code/java/agent/instrumentation/http/#capturing-http-request-and-response-headers)
```

> - Known HTTP methods

Reviewer 2: ```suggestion
- [Known HTTP methods](https://opentelemetry.io/docs/zero-code/java/agent/instrumentation/http/#configuring-known-http-methods)
```

> - Experimental HTTP telemetry

Reviewer 2: ```suggestion
- [Experimental HTTP telemetry](https://opentelemetry.io/docs/zero-code/java/agent/instrumentation/http/#enabling-experimental-http-telemetry)
```

see the
[opentelemetry-spring-webmvc-5.3 instrumentation library](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/spring/spring-webmvc/spring-webmvc-5.3/library).

As it's possible with the
[Java agent](/docs/zero-code/java/agent/configuration), you can configure the

> capture of:

Reviewer: ```suggestion
capture of the following entities:
```
